### PR TITLE
AWS - SES - Create new resource type configuration-set for SES

### PR DESCRIPTION
**Pull request by peterheo**
Creating a new resource type for the SES service.

### CODE REVIEW

File: c7n/resources/ses.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from c7n.utils import local_session
+from c7n.tags import universal_augment
+
+class DescribeConfigurationSet(DescribeSource):
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('ses')
+        for r in resources:
+            details = client.describe_configuration_set(ConfigurationSetName=r['Name'], 
+                ConfigurationSetAttributeNames=['eventDestinations','trackingOptions','deliveryOptions','reputationOptions'])
```
Comment (PratMis) [inline c7n/resources/ses.py:15]: Do we need this at all? Looking at the api, it seems to be a required field and you're trying to feed it all possible values. I feel that we could omit this completely

Comment (PratMis) [reply c7n/resources/ses.py:15]: I see why it's there. Without setting `ConfigurationSetAttributeNames`, we just don't get anything back.
```
>>> response = client.describe_configuration_set(ConfigurationSetName='c7n-test')
>>> response
{'ConfigurationSet': {'Name': 'c7n-test'}, 'ResponseMetadata': {'RequestId': '73aaad6f-6c0b-4978-8521-c291e236e3f7', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Wed, 12 Apr 2023 21:58:30 GMT', 'content-type': 'text/xml', 'content-length': '373', 'connection': 'keep-alive', 'x-amzn-requestid': '73aaad6f-6c0b-4978-8521-c291e236e3f7'}, 'RetryAttempts': 0}}
```

```
>>> response = client.describe_configuration_set(ConfigurationSetName='c7n-test', ConfigurationSetAttributeNames=['deliveryOptions', 'eventDestinations', 'trackingOptions', 'reputationOptions',])
>>> response
{'ConfigurationSet': {'Name': 'c7n-test'}, 'DeliveryOptions': {'TlsPolicy': 'Require'}, 'ReputationOptions': {'SendingEnabled': True, 'ReputationMetricsEnabled': True}, 'ResponseMetadata': {'RequestId': '4bcdf5ed-c74a-43af-8892-ccb2da922b4b', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Wed, 12 Apr 2023 21:58:13 GMT', 'content-type': 'text/xml', 'content-length': '612', 'connection': 'keep-alive', 'x-amzn-requestid': '4bcdf5ed-c74a-43af-8892-ccb2da922b4b'}, 'RetryAttempts': 0}}
```

**Review by PratMis — APPROVED** [review]
lgtm, thanks

**Comment by PratMis** [comment]
@ajkerrigan / @thisisshi, when you get a chance, do you mind taking a look at it one more time. Code looks good to me, I also went ahead and tested the tagging functionality and can confirm that the resource supports resourcegroupstagging api. Thanks :)

File: c7n/resources/ses.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            r.update(details)
```
Comment (ajkerrigan) [inline c7n/resources/ses.py:16]: I think we can ditch a couple keys here. We already have the configuration set name, and we generally don't want to record the response metadata.

```suggestion
            r.update({
                k: details[k]
                for k in details
                if k not in {'ConfigurationSet', 'ResponseMetadata'}
            })
```

**Review by ajkerrigan — CHANGES_REQUESTED** [review]
Thanks, the core looks good to me. Suggestion inline to avoid including noise or duplicate data during augment.

File: c7n/resources/ses.py (hunk 2 above)
Comment (PratMis) [reply c7n/resources/ses.py:16]: oh good point, yup completely missed that.

**Review by PratMis — APPROVED** [review]
Awesome, thanks @peterheo and @ajkerrigan

**Review by ajkerrigan — APPROVED** [review]
Thanks @peterheo and @PratMis :raised_hands:

**Pull request merged**